VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/services/video-call.service.ts

```diff
@@ -237,14 +235,20 @@ export class VideoCallService {
             this.logger.info(`${this.loggerPrefix} Disconnecting from pexip node.`);
             this.stopPresentation();
             this.pexipAPI.disconnect();
-            this.hasDisconnected$.next();
-            this.hasDisconnected$.complete();
-            this.kinlyHeartbeatService.stopHeartbeat();
+            this.cleanUpConnection();
         } else {
             throw new Error(`${this.loggerPrefix} Pexip Client has not been initialised.`);
         }
     }
 
+    private cleanUpConnection() {
+        this.logger.warn(`${this.loggerPrefix} Cleaning up connection.`);
+        this.hasDisconnected$.next();
+        this.hasDisconnected$.complete();
+        this.kinlyHeartbeatService.stopHeartbeat();
+        this.setupClient();
```

Comment:
do we not have access to the instance of the pex rtc and so can we not just set that to null or something rather than setting up client? Or is this done intentionally for some other reason?

Reply:
We need to re run set up client to set up all the event handlers and the streams.

Reply:
but wouldn't that get called anyway when its required? So I am wondering isnt there a function that ensures this instance of pexrtc is cleared?

Reply:
No it doesn't it is only done on event hub is ready. This is the cleanest way. No this is what was advised by Alan (from Pexip)